src/main/java/io/kestra/plugin/jira/issues/JiraClient.java

```diff
@@ -0,0 +1,145 @@
+    title = "Create a jira ticket on a task failure.",
+    description = "Add this task to a list of `errors` tasks to implement custom flow-level failure notifications."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Create a jira ticket on a failed flow execution",
+            full = true,
+            code = """
+                id: unreliable_flow
+                namespace: company.team
+
+                tasks:
+                  - id: fail
+                    type: io.kestra.plugin.scripts.shell.Commands
+                    runner: PROCESS
+                    commands:
+                      - exit 1
+
+                errors:
+                  - id: alert_on_failure
+                    type: io.kestra.plugin.jira.issues.JiraClient
+                    baseUrl: your-domain.atlassian.net
+                    username: your_email@example.com
+                    password: your_jira_api_token
+                    payload: |
+                        {
+                             'fields': {
+                                 'project': {
+                                     'key': PROJECT_KEY
+                                 },
+                                 'summary': ISSUE_SUMMARY,
+                                 'description': ISSUE_DESCRIPTION,
+                                 'issuetype': {
+                                     'name': 'Story'  # Or 'Bug', 'Story', etc.
+                                 },
+                                 'labels': ISSUE_LABELS
+                             }
+                         }
+                    username: your_email@example.com
+                    password: your_jira_api_token
+                """
+        ),
+        @Example(
+            title = "Create a jira ticket",
+            full = true,
+            code = """
+                id: myflow
+                namespace: company.myteam
+
+                tasks:
+                  - id: hello
+                    type: io.kestra.plugin.jira.issues.JiraClient
+                    baseUrl: your-domain.atlassian.net
+                    payload: |
+                        {
+                             'fields': {
+                                 'project': {
+                                     'key': PROJECT_KEY
+                                 },
+                                 'summary': ISSUE_SUMMARY,
+                                 'description': ISSUE_DESCRIPTION,
+                                 'issuetype': {
+                                     'name': 'Story'  # Or 'Bug', 'Story', etc.
+                                 },
+                                 'labels': ISSUE_LABELS
+                             }
+                         }
+                    username: your_email@example.com
+                    password: "{{ secret('your_jira_api_token') }}"
+                 """
+        ),
+    }
+)
+public class JiraClient extends Task implements RunnableTask<VoidOutput> {
```

Comment:
This task is basically an HTTP request task wich we already have and didn't provide any Jira specific behavior.
I think this would be best as an abstract class so users may not be able to use it directly.